forward/forwarder_test.go

```diff
@@ -188,6 +293,85 @@ func (s *ForwarderTestSuite) TestRequestNoReroutes() {
 	s.Error(err)
 }
 
+func (s *ForwarderTestSuite) TestRegisterListener() {
+	listener := &EventListener{}
+	listener.On("HandleEvent").Return()
+
+	s.forwarder.RegisterListener(listener)
+	s.Assertions.Equal(1, len(s.forwarder.listeners), "Expected 1 listener to be registered")
+
+	// remove all listeners
+	s.forwarder.listeners = nil
```

Comment:
Why must this be done? Is there only one test subject (forwarder) created for all tests? That's typically not a great pattern. We should only need to clean up a test subject if there are resources like connections, etc that can leak, but not because to remove listeners.